server/controllers/admin/holidays.js

```diff
@@ -0,0 +1,147 @@
+  var sql =`
+    SELECT h.id, h.label, h.employee_id, h.dateFrom, h.dateTo, h.percentage
+    FROM holiday AS h  
+    WHERE h.id = ?`;
+
+  return db.one(sql, [id]);
+}
+
+// Check Holidays
+/** 
+*
+* This function prevents to define for an employee, two periods of holidays that fits nested
+*/
+function checkHoliday(param) {
+  var sql = `
+    SELECT id, employee_id, label, dateTo, percentage, dateFrom 
+    FROM holiday WHERE employee_id = ?
+    AND ((dateFrom >= DATE(?)) OR (dateTo >= DATE(?)) OR (dateFrom >= DATE(?))
+    OR (dateTo >= DATE(?)))
+    AND ((dateFrom <= DATE(?)) OR (dateTo <= DATE(?)) OR (dateFrom <= DATE(?))
+    OR (dateTo <= DATE(?)))
+  `;
+
+  return db.exec(sql, [param.employee_id, param.dateFrom, param.dateFrom, param.dateTo, param.dateTo, param.dateFrom, param.dateFrom, param.dateTo, param.dateTo]);
+}
+
+
+// Lists the Payroll Holidays
+function list(req, res, next) {
+  const sql = `
+    SELECT h.id, h.label, h.employee_id, h.dateFrom, h.dateTo, p.display_name, h.percentage
+    FROM holiday AS h
+    JOIN employee AS e ON e.id = h.employee_id
+    JOIN patient AS p ON p.uuid = e.patient_uuid
+  ;`;
+
+  db.exec(sql)
+    .then((rows) => {
+      res.status(200).json(rows);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* GET /Holiday/:ID
+*
+* Returns the detail of a single Holiday
+*/
+function detail(req, res, next) {
+  var id = req.params.id;
+
+  lookupHoliday(id)
+    .then((record) => {
+      res.status(200).json(record);
+    })
+    .catch(next)
+    .done();
+}
+
+// POST /Holiday
+function create(req, res, next) {
+  const sql = `INSERT INTO holiday SET ?`;
+  const data = req.body;
+
+  checkHoliday(data)
+    .then((record) => {
+      if (record.length) {
+        throw new Unauthorized('Holiday Nested.', 'ERRORS.HOLIDAY_NESTED');
+      }
+      
+      return db.exec(sql, [data]);
+    })
+    .then((row) => {
+      res.status(201).json({ id : row.insertId });
+    })
+    .catch(next)
+    .done();
+}
+
+
+// PUT /Holiday /:id
+function update(req, res, next) {
+  const sql = `UPDATE holiday SET ? WHERE id = ?;`;
+
+  checkHoliday(req.body)
+    .then((record) => {
+      if (record.length > 1) {
+        throw new Unauthorized('Holiday Nested.', 'ERRORS.HOLIDAY_NESTED');
+      }
+      
+      return db.exec(sql, [req.body, req.params.id]);
+    })  
+    .then(() => {
+      return lookupHoliday(req.params.id);
+    })
+    .then((record) => {
+    // all updates completed successfull, return full object to client
+      res.status(200).json(record);
+    })
+    .catch(next)
+    .done();
+}
+
+// DELETE /Holiday/:id
+function del(req, res, next) {
+  const sql = `DELETE FROM holiday WHERE id = ?;`;
+
+  db.exec(sql, [req.params.id])
+    .then((row) => {
+    // if nothing happened, let the client know via a 404 error
+      if (row.affectedRows === 0) {
+        throw new NotFound(`Could not find a Holiday with id ${req.params.id}`);
+      }
+
+      res.status(204).json();
```

Comment:
It would be better to use `res.sendStatus(204)` since this response isn't actually JSON.